### PR TITLE
Added Youtrack Workflow editor 40.3384

### PR DESCRIPTION
**Pull request by cafferata**
#### Adding a new cask Youtrack Workflow editor 40.3384
- [x] Checked there aren’t open [pull requests](https://github.com/caskroom/homebrew-cask/pulls) for the same cask.
- [x] Checked there aren’t closed [issues](https://github.com/caskroom/homebrew-cask/issues) where that cask was already refused.
- [x] When naming the cask, followed the [token reference](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] Commit message includes cask’s name.
- [x] `brew cask audit --download youtrack-workflow` is error-free.
- [x] `brew cask style --fix youtrack-workflow` left no offenses.
- [x] `brew cask install youtrack-workflow` worked successfully.
- [x] `brew cask uninstall youtrack-workflow` worked successfully.

### CODE REVIEW

File: Casks/youtrack-workflow.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+cask 'youtrack-workflow' do
+  version '6.5.17105'
+  sha256 '331bd2745696507da7ee3a6a193587fc53739face966c12736c16235211cda14'
+
+  # https://download.jetbrains.com/charisma/youtrack-workflow-editor-3384fix-macos.zip.sha256
```
Comment (vitorgalvao) [inline Casks/youtrack-workflow.rb]: This must be removed.

**Comment by vitorgalvao** [comment]
Follow the [deleted cask](https://github.com/caskroom/homebrew-cask/pull/16654/files), for what this is missing.

@adidalal Do you confirm the lack of an app bundle has been resolved?

**Comment by adidalal** [comment]
Version should be `3384fix`

Cask looks OK otherwise.

**Comment by cafferata** [comment]
> Follow the deleted cask, for what this is missing.

I did an diff on the old deleted Youtrack workflow editor cask and added the missing info. The caveats are no longer applicable. I tested the Youtrack Workflow editor with the following Java versions

```bash
java version "1.8.0_92"
Java (TM) SE Runtime Environment (build 1.8.0_92-b14)
Java HotSpot (TM) 64-Bit Server VM (build 25.92-b14, mixed mode)
```

**Comment by vitorgalvao** [comment]
It still requires java, so [`depends_on_java`](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/caveats.md#caveats-mini-dsl) is still necessary.

File: Casks/youtrack-workflow.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+cask 'youtrack-workflow' do
+  version '3384fix'
+  sha256 '331bd2745696507da7ee3a6a193587fc53739face966c12736c16235211cda14'
+
+  name 'Youtrack Workflow Editor'
+  url 'https://download-cf.jetbrains.com/charisma/youtrack-workflow-editor-3384fix-macos.zip'
```
Comment (vitorgalvao) [inline Casks/youtrack-workflow.rb]: `url "https://download-cf.jetbrains.com/charisma/youtrack-workflow-editor-#{version}-macos.zip"`

**Comment by vitorgalvao** [comment]
There was a [pull request template](https://github.com/blog/2111-issue-and-pull-request-templates) when you submitted. It must be filled.

If you were not presented a template, you may have submitted via a CLI tool (likely [`hub`](https://hub.github.com/)). It that was the case, you may wish to see if the tool has a way to let you use the template (`hub` [is working on it](https://github.com/github/hub/issues/1118)).

In either case, please [refer to the template](https://raw.githubusercontent.com/caskroom/homebrew-cask/master/.github/PULL_REQUEST_TEMPLATE.md) and update your comment.

**Comment by cafferata** [comment]
@victorpopkov Thanks for your feedback.

> It still requires java, so depends_on_java is still necessary.

You're right! I added the original caveats.

> In either case, please refer to the template and update your comment.

Done! :-)

**Comment by adidalal** [comment]
@cafferata 

Looks like Java 8 works, so all you actually need is:

```
caveats do
    depends_on_java
end
```

**Comment by cafferata** [comment]
> Looks like Java 8 works, so all you actually need is:

Changed. I tried to deleted all the Java gang. However, removing Java from Mac OS was a lot harder than a single delete command. Thanks again.

File: Casks/youtrack-workflow.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+  url "https://download-cf.jetbrains.com/charisma/youtrack-workflow-editor-#{version}-macos.zip"
+  name 'Youtrack Workflow Editor'
+  homepage 'https://www.jetbrains.com/youtrack/download/get_youtrack.html#materials=workflow-editor'
+  license :gratis
+
+  app 'youtrack-workflow.app'
+
+  caveats do
+      depends_on_java(8)
```
Comment (vitorgalvao) [inline Casks/youtrack-workflow.rb]: Should be just `depends_on_java`. Also, there’s too much indentation. Should be only two spaces.

File: Casks/youtrack-workflow.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+cask 'youtrack-workflow' do
+  version '3384fix'
+  sha256 '331bd2745696507da7ee3a6a193587fc53739face966c12736c16235211cda14'
+
+  url "https://download-cf.jetbrains.com/charisma/youtrack-workflow-editor-#{version}-macos.zip"
+  name 'Youtrack Workflow Editor'
```
Comment (vitorgalvao) [inline Casks/youtrack-workflow.rb]: Should be `name 'JetNrains Youtrack Workflow Editor'`.

Comment (cafferata) [inline Casks/youtrack-workflow.rb]: How's that? It's also not JetBrains [PhpStorm](https://github.com/caskroom/homebrew-cask/blob/master/Casks/phpstorm.rb)?

File: Casks/youtrack-workflow.rb (hunk 3 above)
Comment (cafferata) [inline Casks/youtrack-workflow.rb]: Changed.

File: Casks/youtrack-workflow.rb (hunk 4 above)
Comment (vitorgalvao) [inline Casks/youtrack-workflow.rb]: Not every cask has the correct `name` (fixed that one now), but [it’s the expected](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/name.md).

Comment (cafferata) [inline Casks/youtrack-workflow.rb]: That's making sense. I have changed the name of this Cask.

**Pull request merged**